### PR TITLE
Parse (& Iter 2)

### PR DESCRIPTION
**Pull request by Myrdden**
So it does like parse but also all of iteration 2 and whatever

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    if !@game_stats.has_key?(:home_wins)
+      homeWins = @games.count {|x| x[:home_goals] > x[:away_goals]}.to_f
+      @game_stats[:home_wins] = ((homeWins / @games.count) * 100.0).round(2)
+    end
+    return @game_stats[:home_wins]
+  end
+
+  def percentage_visitor_wins
+    if !@game_stats.has_key?(:away_wins)
+      awayWins = @games.count {|x| x[:away_goals] > x[:home_goals]}.to_f
```
Comment (DavisC0801) [inline lib/game.rb]: Variable name in camel case

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  def biggest_blowout
+    if !@game_stats.has_key?(:difference)
+      @game_stats[:difference] = get_score_differences(@games, []).max
+    end
+    return @game_stats[:difference]
+  end
+
+  def percentage_home_wins
+    if !@game_stats.has_key?(:home_wins)
+      homeWins = @games.count {|x| x[:home_goals] > x[:away_goals]}.to_f
```
Comment (DavisC0801) [inline lib/game.rb]: Variable name in camel case

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    end
+    return @game_stats[:average_game]
+  end
+
+  def average_goals_by_season
+    if !@game_stats.has_key?(:average_season)
+      seasons = @games.map {|x| x[:season]}.uniq
+      @game_stats[:average_season] = {}
+      seasons.each do |season|
+        seasonGames = @games.map {|x| x[:season] == season}
```
Comment (DavisC0801) [inline lib/game.rb]: Variable name in camel case

File: lib/game.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
+class Game
```
Comment (DavisC0801) [inline lib/game.rb:3]: Needs recursion extend and decor rec.

File: lib/game.rb (hunk 2 above)
Comment (DavisC0801) [reply lib/game.rb]:
```suggestion
      home_wins = @games.count {|x| x[:home_goals] > x[:away_goals]}.to_f
```

File: lib/game.rb (hunk 1 above)
Comment (Myrdden) [reply lib/game.rb]: Fixed

File: lib/game.rb (hunk 2 above)
Comment (Myrdden) [reply lib/game.rb]: Fixed

File: lib/game.rb (hunk 3 above)
Comment (Myrdden) [reply lib/game.rb]: Fixed

**Review by DavisC0801 — APPROVED** [review]
Looks good, code was co-created.

**Comment by DavisC0801** [comment]
Merging to master.

**Pull request merged**